src/plugins/data/public/search/session/sessions_mgmt/lib/api.ts

```diff
@@ -35,7 +35,11 @@ function getActions(status: UISearchSessionState) {
     actions.push(ACTION.DELETE);
   }
 
-  if (status === SearchSessionStatus.EXPIRED) {
+  if (
+    status === SearchSessionStatus.EXPIRED ||
+    status === SearchSessionStatus.ERROR ||
+    status === SearchSessionStatus.CANCELLED
+  ) {
```

Comment:
Nit: I think we're adding `ACTION.DELETE` in all cases now, so it can be moved up with the others that fit this category.

Reply:
I think this affects order of the actions in the context menu . 

also, I think if we finish cancelAction (which is implemented already on 90%) then I'd allow to cancel in-progress, but not delete.